src/webkit/wkPage.ts

```diff
@@ -139,6 +141,30 @@ export class WKPage implements PageDelegate {
     });
   }
 
+  onProvisionalLoadStarted(provisionalSession: WKSession) {
+    assert(!this._provisionalPage);
+    this._provisionalPage = new WKProvisionalPage(provisionalSession, this);
+  }
+
+  onProvisionalLoadCommitted(session: WKSession) {
+    assert(this._provisionalPage);
+    assert(this._provisionalPage!._session === session);
+    this._provisionalPage!.commit();
+    this._provisionalPage!.dispose();
+    this._provisionalPage = null;
+    this.setSession(session);
+  }
+
+  onSessionDestroyed(session: WKSession, crashed: boolean) {
+    if (this._provisionalPage && this._provisionalPage._session === session) {
+      this._provisionalPage.dispose();
+      this._provisionalPage = null;
+      return;
+    }
+    if (this._session === session && crashed)
+      this.didClose(crashed);
```

Comment:
Is this the only callsite of didClose?

Reply:
There is still one more in wkPageProxy.ts